habitat-lab/habitat/sims/habitat_simulator/habitat_simulator.py

```diff
@@ -650,9 +667,30 @@ def previous_step_collided(self):
             bool: True if the previous step resulted in a collision, false otherwise
 
         Warning:
-            This feild is only updated when :meth:`step`, :meth:`reset`, or :meth:`get_observations_at` are
-            called.  It does not update when the agent is moved to a new loction.  Furthermore, it
+            This field is only updated when :meth:`step`, :meth:`reset`, or :meth:`get_observations_at` are
+            called.  It does not update when the agent is moved to a new location.  Furthermore, it
             will _always_ be false after :meth:`reset` or :meth:`get_observations_at` as neither of those
             result in an action (step) being taken.
         """
         return self._prev_sim_obs.get("collided", False)
+
+    def add_keyframe_to_observations(self, observations):
```

Comment:
Not a fan of adding batch_rendering specific code to habitat_simulator. But I guess we have to.

Reply:
Another option would be to include this in `habitat-sim`'s `simulator.py` class. However, this relates to functionality that entirely resides within `habitat-lab`, and may cause headaches to the uninitiated.